### PR TITLE
feat(ui): Add recent searches to search bar in Events

### PR DESCRIPTION
**Pull request by billyvg**
This adds recent searches to the Events search bar

image

image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/issueList/searchBar.jsx [hunk 1]
```diff
@@ -47,7 +48,6 @@ class IssueListSearchBar extends React.Component {
     ...SmartSearchBar.propTypes,
 
     savedSearch: SentryTypes.SavedSearch,
-    organization: SentryTypes.Organization.isRequired,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/issueList/searchBar.jsx]: This is now required in `SmartSearchBar`

File: src/sentry/static/sentry/app/components/smartSearchBar/index.jsx [hunk 2]
```diff
@@ -104,9 +103,7 @@ class SmartSearchBar extends React.Component {
   static propTypes = {
     api: PropTypes.object,
 
-    organization: SentryTypes.Organization,
-
-    orgId: PropTypes.string,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/smartSearchBar/index.jsx]: just use organization

File: src/sentry/static/sentry/app/components/smartSearchBar/index.jsx [hunk 3]
```diff
@@ -138,7 +135,13 @@ class SmartSearchBar extends React.Component {
     maxSearchItems: PropTypes.number,
 
     // List user's recent searches
-    displayRecentSearches: PropTypes.bool,
+    hasRecentSearches: PropTypes.bool,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/smartSearchBar/index.jsx]: makes this a bit more consistent with `hasPinnedSearch`

File: src/sentry/static/sentry/app/components/smartSearchBar/index.jsx [hunk 4]
```diff
@@ -266,7 +269,13 @@ class SmartSearchBar extends React.Component {
   };
 
   doSearch = async () => {
-    const {onSearch, onSavedRecentSearch, api, orgId, savedSearchType} = this.props;
+    const {
+      onSearch,
+      onSavedRecentSearch,
+      api,
+      organization,
+      savedSearchType,
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/smartSearchBar/index.jsx]: Where does this get set to a different value for events vs. issues?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/smartSearchBar/index.jsx]: @markstory it comes from props via https://github.com/getsentry/sentry/pull/14420/files#diff-710110490bc7faf582c4f8c1b6eaafcbR87

Comment (billyvg) [reply src/sentry/static/sentry/app/components/smartSearchBar/index.jsx]: It comes from props via https://github.com/getsentry/sentry/pull/14420/files#diff-710110490bc7faf582c4f8c1b6eaafcbR87

**Review by markstory — APPROVED** [review]
Looks pretty good. There is one odd looking diff in percy for user-feedback but that seems like it is likely unrelated.

**Comment by billyvg** [comment]
Re-ran acceptance tests and the user feedback rendered correctly again

**Pull request merged**